### ISSUE TITLE
Map editor crashes

### ISSUE THREAD

**Issue by deniz1a** [OpenRA/OpenRA#9329]
When I open the map editor and load a map, game crashes with the following exception.
````
Tiberian Sun Mod at Version {DEV_VERSION}
Operating System: Windows (Microsoft Windows NT 6.1.7601 Service Pack 1)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Actor does not have trait of type `OpenRA.Mods.Common.Traits.DomainIndex`
   at OpenRA.TraitDictionary.TraitContainer`1.Get(UInt32 actor) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\TraitDictionary.cs:line 148
   at OpenRA.TraitDictionary.Get[T](Actor actor) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\TraitDictionary.cs:line 96
   at OpenRA.Actor.Trait[T]() in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Actor.cs:line 216
   at OpenRA.Mods.Common.AI.HackyAI..ctor(HackyAIInfo info, ActorInitializer init)
   at OpenRA.Mods.Common.AI.HackyAIInfo.Create(ActorInitializer init)
   at OpenRA.Actor..ctor(World world, String name, TypeDictionary initDict) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Actor.cs:line 97
   at OpenRA.World.CreateActor(Boolean addToWorld, String name, TypeDictionary initDict) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\World.cs:line 215
   at OpenRA.World.CreateActor(String name, TypeDictionary initDict) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\World.cs:line 210
   at OpenRA.Player..ctor(World world, Client client, PlayerReference pr) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Player.cs:line 115
   at OpenRA.Mods.Common.Traits.EditorActorLayer.CreatePlayers(World w)
   at OpenRA.World..ctor(Map map, OrderManager orderManager, WorldType type) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\World.cs:line 159
   at OpenRA.Game.StartGame(String mapUID, WorldType type) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Game.cs:line 150
   at OpenRA.Game.LoadEditor(String mapUid) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Game.cs:line 358
   at OpenRA.Mods.Common.Widgets.Logic.MainMenuLogic.<>c__DisplayClass53.<LoadMapIntoEditor>b__51()
   at OpenRA.Mods.Common.Widgets.Logic.ConnectionLogic.ConnectionStateChanged(OrderManager om)
   at System.Action`1.Invoke(T obj)
   at OpenRA.Game.LogicTick() in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Game.cs:line 494
   at OpenRA.Game.Loop() in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Game.cs:line 621
   at OpenRA.Game.Run() in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Game.cs:line 661
   at OpenRA.Program.Run(String[] args) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Support\Program.cs:line 117
   at OpenRA.Program.Main(String[] args) in d:\Dev\openra_dev\OpenRA\OpenRA.Game\Support\Program.cs:line 39

````

**Comment by clemty** [comment]
seeing HackyAI in that log, maybe it is already addressed in #9325

**Comment by Mailaender** [comment]
Yes, it is.
